docs/en/sql-reference/functions/geo/s2.md

```diff
@@ -0,0 +1,396 @@
+
+Determines if a given rectangle contains a S2 point index.
+
+**Syntax**
+
+``` sql
+s2RectContains(s2PointLow, s2PointHi, s2Point)
+```
+
+**Arguments**
+
+- `s2PointLow`  - Low S2 point index corresponding to the rectangle. [UInt64](../../../sql-reference/data-types/int-uint.md).
+- `s2PointHigh` - High S2 point index corresponding to the rectangle. [UInt64](../../../sql-reference/data-types/int-uint.md).
+- `s2Point`     - Target S2 point index. [UInt64](../../../sql-reference/data-types/int-uint.md).
+
+**Returned values**
+
+-   1 — If the rectangle contains the given S2 point.
+-   0 — If the rectangle doesn't contain the given S2 point.
+
+**Example**
+
+Query:
+
+``` sql
+SELECT s2RectContains(5179062030687166815, 5177056748191934217, 5177914411069187297) AS rectContains
+```
+
+Result:
+
+``` text
+┌─rectContains─┐
+│            0 │
+└──────────────┘
+```
+
+## s2RectUinion{#s2RectUnion}
+
+In the S2 system, a rectangle is represented by a type of S2Region called a S2LatLngRect that represents a rectangle in latitude-longitude space.
+
+Returns the smallest rectangle containing the union of this rectangle and the given rectangle.
+
+**Syntax**
+
+``` sql
+s2RectUnion(s2Rect1PointLow, s2Rect1PointHi, s2Rect2PointLow, s2Rect2PointHi)
+```
+
+**Arguments**
+
+- `s2Rect1PointLow`, `s2Rect1PointHi` - Low and High S2 point indices corresponding to the first rectangle. [UInt64](../../../sql-reference/data-types/int-uint.md).
+- `s2Rect2PointLow`, `s2Rect2PointHi` - Low and High S2 point indices corresponding to the second rectangle. [UInt64](../../../sql-reference/data-types/int-uint.md).
+
+**Returned values**
+
+- `s2UnionRect2PointLow` - Low S2 cell id corresponding to the union rectangle. Type: [UInt64](../../../sql-reference/data-types/int-uint.md).
+- `s2UnionRect2PointHi` -  High S2 cell id corresponding to the union rectangle. Type: [UInt64](../../../sql-reference/data-types/int-uint.md).
+
+
+**Example**
+
+Query:
+
+``` sql
+SELECT s2RectUnion(5178914411069187297, 5177056748191934217, 5179062030687166815, 5177056748191934217) AS rectUnion
+```
+
+Result:
+
+``` text
+┌─rectUnion─────────────────────────────────┐
+│ (5179062030687166815,5177056748191934217) │
+└───────────────────────────────────────────┘
+```
+
+## s2RectIntersection{#s2RectIntersection}
+
+Returns the smallest Rectangle containing the intersection of this rectangle and the given rectangle.
+
+**Syntax**
+
+``` sql
+s2RectIntersection(s2Rect1PointLow, s2Rect1PointHi, s2Rect2PointLow, s2Rect2PointHi)
+```
+
+**Arguments**
+
+- `s2Rect1PointLow`, `s2Rect1PointHi` - Low and High S2 point indices corresponding to the first rectangle. [UInt64](../../../sql-reference/data-types/int-uint.md).
+- `s2Rect2PointLow`, `s2Rect2PointHi` - Low and High S2 point indices corresponding to the second rectangle. [UInt64](../../../sql-reference/data-types/int-uint.md).
```

Comment:
Maybe better to say something about this point. E.g. bottom left or top right point of a rectangle ?

Reply:
Seems this is not applicable to s2 points. Maybe say what s2 point is ?

Reply:
Sure, added the definition of what a s2 point index is at the top of the document since it's used in multiple places including this one.